vars/nodeJsBuild.groovy

```diff
@@ -0,0 +1,23 @@
+import com.sap.piper.DownloadCacheUtils
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+import static groovy.json.JsonOutput.toJson
+
+@Field String STEP_NAME = getClass().getName()
+@Field String METADATA_FILE = 'metadata/jsBuild.yaml'
+
+//Metadata maintained in file project://resources/metadata/jsBuild.yaml
+
+void call(Map parameters = [:]) {
+    final script = checkScript(this, parameters) ?: this
+
+    // No credentials required/supported as of now
+    List credentials = []
+
+    parameters['dockerOptions'] = DownloadCacheUtils.getDockerOptions(script)
+    if (DownloadCacheUtils.isEnabled()) {
+        parameters['defaultNpmRegistry'] = DownloadCacheUtils.getNpmRegistryUri(script)
+    }
```

Comment:
Doing it this way, without resolving the config in the Groovy-layer, means this parameter could not be provided via config. In the go-metadata-yaml, you declare the scope for defaultNpmRegistry as PARAMETERS, STAGES and STEPS. But actually the parameter here would always take precedence. If it is really intended to come also via config, then I can't think of a non-ugly solution at the moment, otherwise you could reduce the scope to PARAMETERS in the yaml.

Reply:
Yes I think this should also be configurable in non-jenkins use-cases. So if I understand your comment correctly, you don't see a better alternative to do this?